# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/patches/pkg_e2fsprogs.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./lib/ss/pager.c.orig	2006-06-21 15:44:59.000000000 +0200
+++ ./lib/ss/pager.c	2006-06-21 15:45:54.000000000 +0200
@@ -36,6 +36,9 @@
 #if (!defined(HAVE_PRCTL) && defined(linux))
 #include <sys/syscall.h>
 #endif
+#ifndef PR_GET_DUMPABLE
+#define PR_GET_DUMPABLE 3
+#endif
 
 static char MORE[] = "more";
 extern char *_ss_pager_name;
--- ./lib/ext2fs/test_io.c.orig	2006-06-21 15:51:20.000000000 +0200
+++ ./lib/ext2fs/test_io.c	2006-06-21 15:51:47.000000000 +0200
@@ -34,6 +34,10 @@
 #include "ext2_fs.h"
 #include "ext2fs.h"
 
+#ifndef PR_GET_DUMPABLE
+#define PR_GET_DUMPABLE 3
+#endif
+
 /*
  * For checking structure magic numbers...
  */
--- ./lib/blkid/cache.c.orig	2006-06-21 15:57:47.000000000 +0200
+++ ./lib/blkid/cache.c	2006-06-21 15:58:16.000000000 +0200
@@ -28,6 +28,10 @@
 #endif
 #include "blkidP.h"
 
+#ifndef PR_GET_DUMPABLE
+#define PR_GET_DUMPABLE 3
+#endif
+
 int blkid_debug_mask = 0;
 
 
--- ./lib/ext2fs/ext2_types.h.in.orig	2006-07-08 00:36:04.000000000 +0000
+++ ./lib/ext2fs/ext2_types.h.in	2006-07-08 00:37:06.000000000 +0000
@@ -3,6 +3,8 @@
  * everything we need.  (cross fingers)  Other header files may have 
  * also defined the types that we need.
  */
+#include <linux/types.h>
+
 #if (!defined(_LINUX_TYPES_H) && !defined(_BLKID_TYPES_H) && !defined(_PPC64_TYPES_H) && \
 	!defined(_EXT2_TYPES_H))
 #define _EXT2_TYPES_H
--- ./lib/blkid/blkid.h.orig	2006-07-08 00:41:20.000000000 +0000
+++ ./lib/blkid/blkid.h	2006-07-08 00:41:38.000000000 +0000
@@ -14,6 +14,7 @@
 #define _BLKID_BLKID_H
 
 #include <sys/types.h>
+#include <linux/types.h>
 #include <blkid/blkid_types.h>
 
 #ifdef __cplusplus
--- ./misc/util.c.orig	2006-07-08 00:58:39.000000000 +0000
+++ ./misc/util.c	2006-07-08 00:59:42.000000000 +0000
@@ -79,7 +79,7 @@
 void check_plausibility(const char *device)
 {
 	int val;
-#ifdef HAVE_OPEN64
+#ifndef __NO_STAT64
 	struct stat64 s;
 	
 	val = stat64(device, &s);
